### DISCUSSION TITLE
Swipeable Drawer and Tabs

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [chakra-ui/chakra-ui#5205] [Ideas]
### Description

Swiping on mobile devices is very common and has a better UX.  

### Problem Statement/Justification

support mobile devices with modern actions and reactions.

### Proposed Solution or API

[swipeable drawer](https://github.com/mui-org/material-ui/tree/8cc4abee21aabfb126933dc475745e023323f53f/packages/mui-material/src/SwipeableDrawer)
[swipeable tabs](https://github.com/mui-org/material-ui/blob/512896973499adbbda057e7f3685d1b23cc02de9/docs/src/pages/components/tabs/FullWidthTabs.js)

### Alternatives

probably is a bad idea, using MUI with Chakra

### Additional Information

[swipeable tabs](https://mui.com/components/tabs/#full-width)
[swipeable drawer](https://mui.com/components/drawers/#swipeable)

**Comment by segunadebayo** [comment]
Moved this to a discussion to allow others to upvote this request and leave comments.

**Comment by tombohub** [comment]
+1 gotta focus on mobile, there is no proper UI library for mobile except framework7 and antd mobile. 
There's an opportunity

**Comment by Southclaws** [comment]
Huge +1 here, I did manage to get something basic working with Framer motion's drag attribute combined with Chakra's drawer:

```tsx
import {
  Button,
  Drawer,
  DrawerBody,
  DrawerCloseButton,
  DrawerContent,
  DrawerFooter,
  DrawerHeader,
  DrawerOverlay,
  Input,
  Text,
  useDisclosure,
} from "@chakra-ui/react";

import { motion } from "framer-motion";
import React from "react";

export default function Details() {
  const { isOpen, onOpen, onClose } = useDisclosure();
  const btnRef = React.useRef();

  return (
    <>
      <Button ref={btnRef as any} colorScheme="teal" onClick={onOpen}>
        Open
      </Button>
      <Drawer
        isOpen={isOpen}
        placement="bottom"
        onClose={onClose}
        finalFocusRef={btnRef as any}
      >
        <DrawerOverlay zIndex={110} />
        <motion.div
          style={{
            width: "100%",
            // height: 150,
            borderRadius: 30,
            backgroundColor: "#fff",
            zIndex: 120,
          }}
          drag="y"
          onDragEnd={(e) => {
            console.log(e);
            onClose();
          }}
          dragConstraints={{
            top: 0,
            bottom: 125,
          }}
          dragTransition={{ bounceStiffness: 600, bounceDamping: 20 }}
          dragElastic={0.5}
          //   whileTap={{ cursor: "grabbing" }}
        >
          <DrawerContent borderTopRadius={12}>
            <DrawerCloseButton />
            <DrawerHeader>Drawer</DrawerHeader>

            <DrawerBody>
              <Text>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Qui
                quam beatae sunt adipisci commodi. Nihil earum doloremque
                impedit ipsa. Dignissimos, quo soluta. Ea optio ipsum provident
                itaque numquam perspiciatis in! Lorem ipsum dolor sit amet
                consectetur
              </Text>

              <Text>
                adipisicing elit. Molestiae earum nemo esse, sequi culpa odio et
                fugiat deleniti velit at ea modi magnam facilis adipisci officia
                iusto nam, unde architecto!
              </Text>
            </DrawerBody>
          </DrawerContent>
        </motion.div>
      </Drawer>
    </>
  );
}
```


https://user-images.githubusercontent.com/1636971/200168368-8eeee026-990c-406f-9e6b-486337d57363.mov

**Reply by tombohub** [reply]
nice, what software you used to record screen?

**Reply by Southclaws** [reply]
just the built-in mac screenshot app

**Reply by Lukasvo** [reply]
Great suggestion, thanks @Southclaws!

Wrapping DrawerContent in a `motion.div` messed up my lay-out a bit.
I solved it by doing `<DrawerContent as={motion.div} {...restOfTheMotionProps}>`

Edit: it even works without `as={...}` and just with the motion props!

**Reply by Southclaws** [reply]
What I do nowadays is use https://github.com/emilkowalski/vaul and wrap the modal in that so it's responsive and becomes a modal on bigger screens.
